hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/scheduledexecutor/ScheduledExecutorTaskCancelFromPartitionMessageTask.java

```diff
@@ -0,0 +1,83 @@
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.impl.protocol.task.scheduledexecutor;
+
+import com.hazelcast.client.impl.protocol.ClientMessage;
+import com.hazelcast.client.impl.protocol.codec.ScheduledExecutorCancelFromPartitionCodec;
+import com.hazelcast.client.impl.protocol.task.AbstractPartitionMessageTask;
+import com.hazelcast.instance.Node;
+import com.hazelcast.nio.Connection;
+import com.hazelcast.scheduledexecutor.ScheduledTaskHandler;
+import com.hazelcast.scheduledexecutor.impl.DistributedScheduledExecutorService;
+import com.hazelcast.scheduledexecutor.impl.ScheduledTaskHandlerImpl;
+import com.hazelcast.scheduledexecutor.impl.operations.CancelTaskOperation;
+import com.hazelcast.security.permission.ActionConstants;
+import com.hazelcast.security.permission.ScheduledExecutorPermission;
+import com.hazelcast.spi.Operation;
+
+import java.security.Permission;
+
+public class ScheduledExecutorTaskCancelFromPartitionMessageTask
+        extends AbstractPartitionMessageTask<ScheduledExecutorCancelFromPartitionCodec.RequestParameters> {
+
+    public ScheduledExecutorTaskCancelFromPartitionMessageTask(ClientMessage clientMessage, Node node, Connection connection) {
+        super(clientMessage, node, connection);
+    }
+
+    @Override
+    protected Operation prepareOperation() {
+        ScheduledTaskHandler handler = ScheduledTaskHandlerImpl.of(clientMessage.getPartitionId(),
+                parameters.schedulerName,
+                parameters.taskName);
+        return new CancelTaskOperation(handler, parameters.mayInterruptIfRunning);
+    }
+
+    @Override
+    protected ScheduledExecutorCancelFromPartitionCodec.RequestParameters decodeClientMessage(ClientMessage clientMessage) {
+        return ScheduledExecutorCancelFromPartitionCodec.decodeRequest(clientMessage);
+    }
+
+    @Override
+    protected ClientMessage encodeResponse(Object response) {
+        return ScheduledExecutorCancelFromPartitionCodec.encodeResponse((Boolean) response);
+    }
+
+    @Override
+    public String getServiceName() {
+        return DistributedScheduledExecutorService.SERVICE_NAME;
+    }
+
+    @Override
+    public Permission getRequiredPermission() {
+        return new ScheduledExecutorPermission(parameters.schedulerName, ActionConstants.ACTION_MODIFY);
+    }
+
+    @Override
+    public String getDistributedObjectName() {
+        return parameters.schedulerName;
+    }
+
+    @Override
+    public String getMethodName() {
+        return "cancel";
+    }
+
+    @Override
+    public Object[] getParameters() {
+        return new Object[] {parameters.mayInterruptIfRunning };
```

Comment:
Security interceptors target our public interfaces so for this ClientMessageTask it targets 
`boolean cancel(boolean mayInterruptIfRunning);` of `IScheduledFuture<V>`